src/python/pants/backend/native/tasks/link_shared_libraries.py

```diff
@@ -28,12 +29,15 @@ class LinkSharedLibraryRequest(datatype([
     'object_files',
     'native_artifact',
     'output_dir',
+    # This may be None!
     'external_libs_info'
 ])): pass
```

Comment:
Consider over-riding `__new__` and having it take `external_libs_info=None` to make this more clear - or maybe provide a factory @classmethod to do said same.

Reply:
Done (with several other changes along those lines) in 643417e!